bam_to_mate_hist.py

```diff
@@ -149,19 +164,71 @@ def calc_n50_from_header(header, xx=50.0):
     cumsum = 0
     contig_len = 0
     for length in reversed(lens):
-        contig_len = length
+        contig_len = int(length)
         cumsum += length
         if cumsum >= nxx_len:
             break
-    return contig_len, total
+    greater_10k = [contig for contig in lens if contig > 1e4]
```

Comment:
Since we aren't using the resulting list comprehension for anything, just change `greater_10k` to use the same convention as `above_10k` above:

`greater_10k = len([contig for contig in lens if contig > 1e4])`

and then return `greater_10k` instead of its `len`.

Also, incidentally since your list `lens` is full of ints already, there is what I think will be a faster way to do this if you'd like to switch:

`greater_10k = sum(x > 1e4 for x in lens)`

(The `x > 1e4` makes the comprehension be full of 1's and 0's, so summing them gives you a count, not the sum of `lens`)